### PR TITLE
th-82: Create badge component

### PR DESCRIPTION
**Pull request by albsbz**
Badges
Added badge component that accepts color from pallete and use contrast text color. To use SCSS loops and import scss variables in React component added some changes to stylelint.

### CODE REVIEW

File: frontend/src/.stylelintrc.yml [hunk 1]
```diff
@@ -6,18 +6,25 @@ plugins:
...
+    - true
+    - ignoreSelectors:
+        - :export
   # allow @use to precede @import
   no-invalid-position-at-import-rule:
     - true
     - ignoreAtRules:
         - use
   # allow :global which is specific pseudo class, available in .module.css
+  # allow :export which is specific pseudo class, available in .module.css
```
Comment (Some14u) [inline frontend/src/.stylelintrc.yml]: export isn't specific pseudo class for .module.css
It was my comment about ":global" pseudo, which is actually is part of .module.css logic and by default is unknown to css, scss and stylelint.

This is only comment, so you don't really have to change anything, but if you would have a chance, please, fix it :)

Comment (albsbz) [reply frontend/src/.stylelintrc.yml]: fixed

File: frontend/src/libs/components/badge/badge.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  className?: string;
+  color?: Color;
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+};
+
+const Badge: React.FC<Properties> = ({
+  children,
+  className,
+  color = defaultColor,
+  onClick: handleClick,
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/badge/badge.tsx]: Not necessary to rename handler here. We should name it with `handle` prefix only during declaration.

File: package-lock.json [hunk 3]
```diff
@@ -10511,6 +10496,21 @@
         "stylelint": "^14.0.0 || ^15.0.0"
       }
     },
+    "node_modules/stylelint-scss": {
```
Comment (PavloSukhinin) [inline package-lock.json]: Not sure why package-lock was changed

File: frontend/src/libs/components/badge/badge.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}: Properties): JSX.Element => {
+  const isEvent = Boolean(handleClick);
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/badge/badge.tsx]:
```suggestion
  const hasEvent = Boolean(handleClick);
```

File: frontend/src/libs/components/badge/badge.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  const isColorInPalette = Boolean(paletteColors[color]);
+  const backgroundColor = isColorInPalette ? color : defaultColor;
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/badge/badge.tsx]:
```suggestion
  const backgroundColor = paletteColors[color] ?? defaultColor;
```
to simplify it a bit

File: frontend/src/libs/components/badge/badge.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  const isBackgroundDark = Boolean(darkColors[backgroundColor]);
+  const isClassName = Boolean(className);
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/badge/badge.tsx]:
```suggestion
  const hasClassName = Boolean(className);
```

**Review by PavloSukhinin — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/badge/badge.tsx (hunk 2 above)
Comment (albsbz) [reply frontend/src/libs/components/badge/badge.tsx]: fixed

File: package-lock.json (hunk 3 above)
Comment (albsbz) [reply package-lock.json]: removed

File: frontend/src/libs/components/badge/badge.tsx (hunk 4 above)
Comment (albsbz) [reply frontend/src/libs/components/badge/badge.tsx]: fixed

File: frontend/src/libs/components/badge/badge.tsx (hunk 5 above)
Comment (albsbz) [reply frontend/src/libs/components/badge/badge.tsx]: fixed with  `const backgroundColor = paletteColors[color] ? color : defaultColor;`

File: frontend/src/libs/components/badge/badge.tsx (hunk 6 above)
Comment (albsbz) [reply frontend/src/libs/components/badge/badge.tsx]: fixed

**Review by PavloSukhinin — DISMISSED** [review]

**Review by Druxight — DISMISSED** [review]

**Review by PavloSukhinin — APPROVED** [review]

**Pull request merged**